litestar/serialization/msgspec_hooks.py

```diff
@@ -98,44 +88,48 @@ def default_serializer(value: Any, type_encoders: Mapping[Any, Callable[[Any], A
     raise TypeError(f"Unsupported type: {type(value)!r}")
 
 
-def dec_hook(type_: Any, value: Any) -> Any:  # pragma: no cover
+def default_deserializer(
+    target_type: Any, value: Any, type_decoders: TypeDecodersSequence | None = None
+) -> Any:  # pragma: no cover
     """Transform values non-natively supported by ``msgspec``
 
     Args:
-        type_: Encountered type
+        target_type: Encountered type
         value: Value to coerce
+        type_decoders: Optional sequence of type decoders
 
     Returns:
         A ``msgspec``-supported type
     """
 
     from litestar.datastructures.state import ImmutableState
 
-    if isinstance(value, type_):
+    if isinstance(value, target_type):
         return value
 
-    for predicate, decoder in EXTRA_DECODERS:
-        if predicate(type_):
-            return decoder(type_, value)
+    if type_decoders:
+        for predicate, decoder in type_decoders:
+            if predicate(target_type):
+                return decoder(target_type, value)
 
-    if issubclass(type_, (Path, PurePath, ImmutableState, UUID)):
-        return type_(value)
+    if issubclass(target_type, (Path, PurePath, ImmutableState, UUID)):
+        return target_type(value)
 
     raise TypeError(f"Unsupported type: {type(value)!r}")
 
 
 _msgspec_json_encoder = msgspec.json.Encoder(enc_hook=default_serializer)
-_msgspec_json_decoder = msgspec.json.Decoder(dec_hook=dec_hook)
+_msgspec_json_decoder = msgspec.json.Decoder(dec_hook=default_deserializer)
 _msgspec_msgpack_encoder = msgspec.msgpack.Encoder(enc_hook=default_serializer)
-_msgspec_msgpack_decoder = msgspec.msgpack.Decoder(dec_hook=dec_hook)
+_msgspec_msgpack_decoder = msgspec.msgpack.Decoder(dec_hook=default_deserializer)
 
 
-def encode_json(obj: Any, default: Callable[[Any], Any] | None = None) -> bytes:
+def encode_json(value: Any, serializer: Callable[[Any], Any] | None = None) -> bytes:
     """Encode a value into JSON.
 
     Args:
-        obj: Value to encode
-        default: Optional callable to support non-natively supported types.
+        value: Value to encode
+        serializer: Optional callable to support non-natively supported types.
```

Comment:
i changes the name throughout - we are leaking implementation details IMO, and its not a very nice convention.